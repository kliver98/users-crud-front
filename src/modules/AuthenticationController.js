import axios from 'axios';

const baseAPIauth = process.env.VUE_APP_BASE_URL_API_AUTH

export const AuthenticationController = {
    authenticate(credentials) {
        return axios({
            method: 'POST',
            baseURL: baseAPIauth,
            url: 'users/auth/',
            data: credentials,
        })
    }
}
