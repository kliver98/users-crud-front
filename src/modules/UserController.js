import axios from 'axios';

const baseAPIuser = process.env.VUE_APP_BASE_URL_API_USER

export const UserController = {
    create(user) {
        return axios({
            method: 'POST',
            baseURL: baseAPIuser,
            url: 'users',
            data: user,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: baseAPIuser,
            url: 'users'
        });
    },
    delete(id) {
        return axios({
            method: 'DELETE',
            baseURL: baseAPIuser,
            url: `users/${id}`,
        });
    },
    getUser(id) {
        return axios({
            method: 'GET',
            baseURL: baseAPIuser,
            url: `users/${id}`,
        });
    },
    updateUser(id, user) {
        return axios({
            method: 'PUT',
            baseURL: baseAPIuser,
            url: `users/${id}`,
            data: user,
        });
    }
}
