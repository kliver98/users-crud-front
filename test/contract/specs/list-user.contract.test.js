import { provider } from '../config/init-pact-user';
import { UserController as controller } from '../../../src/modules/UserController';
import { Matchers } from '@pact-foundation/pact';

describe('Given an user service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to list all the users is made', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'has users and want to list them',
                uponReceiving: 'a request to list all users',
                withRequest: {
                    method: 'GET',
                    path: '/users'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        name: Matchers.string('Manchas'),
                        breed: Matchers.like("Bengali"),
                        gender: Matchers.like("Female"),
                        vaccinated: Matchers.boolean(true),
                    }, {min: 1})
                }
            });
        });

        test('Then it should return the right data', async() => {
            const response = await controller.list();
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
