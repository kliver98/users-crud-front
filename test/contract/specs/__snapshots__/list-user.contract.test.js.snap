// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an user service When a request to list all the users is made Then it should return the right data 1`] = `
Array [
  Object {
    "_id": 0,
    "username": "admin",
  },
]
`;
